zerver/worker/queue_processors.py

```diff
@@ -224,7 +224,10 @@ def consume(self, data: Mapping[str, Any]) -> None:
 
         # queue invitation reminder
         if settings.INVITATION_LINK_VALIDITY_DAYS >= 4:
-            link = create_confirmation_link(invitee, referrer.realm.host, Confirmation.INVITATION)
+            confirmation = Confirmation.objects.filter(object_id=invitee.id,
+                                                       type=Confirmation.INVITATION).latest('date_sent')
+            link = confirmation_url(confirmation.confirmation_key, referrer.realm.host,
+                                    Confirmation.INVITATION)
```

Comment:
Nice find!

A cleaner approach would be to just have `do_send_confirmation_email` return the URL; I'll merge this with that change.